File: backend/Models/Domain/Teacher.cs
namespace _3w1m.Models.Domain;

public class Teacher  
{
    public int TeacherId { get; set; }
    
    public string Fullname { get; set; }
    
    public string ContactEmail { get; set; }
    
    public string ContactPhone { get; set; }
    
    public string? UserId { get; set; }

    public User User { get; set; }
}

File: frontend/src/types/teacher.ts
export default interface Teacher {
  id: string;
  fullname: string;
  contactEmail: string;
  contactPhone: string;
}

export interface TeacherCollectionResponse {
  TeacherId: string;
  Fullname: string;
  ContactEmail: string;
  ContactPhone: string;
  UserId: string;
}

export interface CreateTeacherDTO {
  fullname: string;
  password: string;
  contactEmail: string;
  contactPhone: string;
}

export interface UpdateTeacherDTO {
  fullname?: string;
  contactEmail?: string;
  contactPhone?: string;
}

export interface TeacherResponse {
  TeacherId: string;
Fullname: string;
}
